File: CS/CustomControlExample/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace TreeListExample
{
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport
    {
        public XtraReport1()
        {
            InitializeComponent();
        }

        private void xrTreeList1_PrintNode(object sender, DevExpress.XtraReports.CustomControls.PrintNodeEventArgs e)
        {
            //if ((int)e.Node.KeyValue == 2)
            //    e.SuppressType = DevExpress.XtraReports.CustomControls.NodeSuppressType.Suppress;
            //if ((int)e.Node.KeyValue == 21)
            //    e.SuppressType = DevExpress.XtraReports.CustomControls.NodeSuppressType.SuppressWithChildren;
        }

        private void xrTreeList1_PrintNodeCell(object sender, DevExpress.XtraReports.CustomControls.PrintNodeCellEventArgs e)
        {
            //if ((int)e.Node.KeyValue == 2)
            //    e.Style.BackColor = Color.Red;
        }

        private void xrTreeList1_PrintHeaderCell(object sender, DevExpress.XtraReports.CustomControls.PrintCellEventArgs e)
        {
            //e.Style.BackColor = Color.Aqua;
        }

    }
}

File: VB/CustomControlExample/XtraReport1.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace TreeListExample
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub xrTreeList1_PrintNode(ByVal sender As Object, ByVal e As DevExpress.XtraReports.CustomControls.PrintNodeEventArgs)
			'if ((int)e.Node.KeyValue == 2)
			'    e.SuppressType = DevExpress.XtraReports.CustomControls.NodeSuppressType.Suppress;
			'if ((int)e.Node.KeyValue == 21)
			'    e.SuppressType = DevExpress.XtraReports.CustomControls.NodeSuppressType.SuppressWithChildren;
		End Sub

		Private Sub xrTreeList1_PrintNodeCell(ByVal sender As Object, ByVal e As DevExpress.XtraReports.CustomControls.PrintNodeCellEventArgs)
			'if ((int)e.Node.KeyValue == 2)
			'    e.Style.BackColor = Color.Red;
		End Sub

		Private Sub xrTreeList1_PrintHeaderCell(ByVal sender As Object, ByVal e As DevExpress.XtraReports.CustomControls.PrintCellEventArgs)
			'e.Style.BackColor = Color.Aqua;
		End Sub

	End Class
End Namespace
